compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/clauses/BLangMatchClause.java

```diff
@@ -89,13 +89,13 @@ public void addMatchPattern(MatchPatternNode matchPattern) {
     }
 
     @Override
-    public boolean isLastClause() {
-        return isLastClause;
+    public boolean hasLastPatternInClause() {
+        return hasLastPatternInClause;
     }
 
     @Override
-    public void setLastClause() {
-        this.isLastClause = true;
+    public void setLastPatternInClause() {
+        this.hasLastPatternInClause = hasLastPatternInClause;
```

Comment:
Shouldn't this be
```suggestion
        this.hasLastPatternInClause = true;
```

Is `setLastPatternInClause` not used btw? If it is used, there should have been tests that failed due to this, right?

Reply:
Removed that setter